dateutil/parser.py

```diff
@@ -732,11 +732,16 @@ def _parse(self, timestr, dayfirst=None, yearfirst=None, fuzzy=False,
                             res.second, res.microsecond = _parsems(s[4:])
 
                     elif len_li in (8, 12, 14):
-                        # YYYYMMDD
+                        # YYYYMMDD or DDMMYYYY
                         s = l[i-1]
-                        ymd.append(s[:4])
-                        ymd.append(s[4:6])
-                        ymd.append(s[6:8])
+                        if int(s[4:6]) <= 12 and int(s[6:8]) <= 31:
```

Comment:
Hm... I think the logic on this part of the parser is not great anyway, so we don't have to worry about this so much, but consider:

```python
>>> from dateutil.parser import parse
>>> parse('20050931')
Traceback: [...]
ValueError: day is out of range for month
```

This means the parser would fail for the unambiguous date '20050931', which is unambiguously May 20, 931. That said, since *all* the dates surrounding May 20, 931 would parse incorrectly if rendered in this format, this might be a reasonable step to take. There are various simple ways to mitigate this, but I don't think they're worth doing at the moment, I think most of this logic is going to get redone int he end.